main.py

```diff
@@ -33,42 +40,57 @@ def register_input():
             if event.key == pygame.K_DOWN:
                 a[2] = 0
 
+
 # Create the pygame screen, used for debugging.
 screen = pygame.display.set_mode((SCREEN_WIDTH, SCREEN_HEIGHT))
 
 # Create the environment.
 env = gym.make("CarRacing-v2", max_episode_steps=MAX_STEPS)
 obs, info = env.reset()
 
+
 while True:
     register_input()
     observation, reward, terminated, truncated, info = env.step(a)
 
     # Render the observation using pygame.
     # Rotate + flip the observation, see: 
     # https://stackoverflow.com/questions/66241275/pygame-rotates-camera-stream-from-opencv-camera.
-    rotated = np.rot90(observation, k=1, axes=(0,1))
+    rotated = np.rot90(observation, k=1, axes=(0, 1))
     flipped = np.flip(rotated, axis=0)
     extracted_speed = feature_extraction.extract_true_speed(observation)
     extracted_abs = feature_extraction.extract_abs(observation)
+    indicator_bar = feature_extraction.extract_indicators(flipped)
```

Comment:
Why not use the observation?

Reply:
Fine, I will change this to use observation